hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/nfttokeninfo/NftTokenInfoCall.java

```diff
@@ -90,11 +93,27 @@ public NftTokenInfoCall(
         }
 
         final var nonNullNft = nft != null ? nft : Nft.DEFAULT;
+        final var ownerAccount = getOwnerAccount(nonNullNft, token);
+        if (ownerAccount == null) {
+            return revertResult(INVALID_ACCOUNT_ID, gasCalculator.viewGasRequirement());
+        }
         return successResult(
                 NON_FUNGIBLE_TOKEN_INFO
                         .getOutputs()
                         .encodeElements(
-                                status.protoOrdinal(), nftTokenInfoTupleFor(token, nonNullNft, serialNumber, ledgerId)),
+                                status.protoOrdinal(),
+                                nftTokenInfoTupleFor(token, nonNullNft, serialNumber, ledgerId, ownerAccount)),
                 gasRequirement);
     }
+
+    private Account getOwnerAccount(Nft nft, Token token) {
```

Comment:
Since we don't have unit tests for this new code, I think we should make the relevant logic in `OwnerOfCall` [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/ownerof/OwnerOfCall.java#L57) a `static` helper and re-use that code here instead of duplicating it.

Reply:
Updated it, thank you :)